packages/contracts/contracts/plugin/PluginSetupProcessor.sol

```diff
@@ -198,8 +198,14 @@ contract PluginSetupProcessor is DaoAuthorizable {
     /// @notice Emitted after a plugin installation was applied.
     /// @param dao The address of the dao to which the plugin belongs.
     /// @param plugin The address of the plugin contract.
-    /// @param setupId TOD:GIORGI: The setup Id hash of the plugin's dependencies.
-    event InstallationApplied(address indexed dao, address indexed plugin, bytes32 setupId);
+    /// @param preparedSetupId TOD:GIORGI: The setup Id hash of the plugin's dependencies.
+    /// @param installationSetupId TOD:GIORGI: The setup Id hash of the plugin's dependencies.
+    event InstallationApplied(
+        address indexed dao,
+        address indexed plugin,
+        bytes32 preparedSetupId,
+        bytes32 installationSetupId
```

Comment:
These are different IDs with different purposes.
Yet the result from the same function where some of the arguments are nulled out.
`installationSetupId`
`updateSetupId`
exist to be consumed by future setups (e.g., future updates or uninstallations).
Yet, they are calculated with the same `getSetupId` function, although they are not a real `setupId`s.

`uninstallationSetupId` is just `bytes32(0)`

Don't you think it makes sense to create a second function (besides `getSetupId`) that takes less argument and to give the ID a different name?

Reply:
I'd need to think a little more to bring this change to be honest, but I agree that you could create one to make things simpler. @brickpop What would you say ? shall we spend some time on it to dive into it ?

Reply:
+1 on the potentially confusing names

Reply:
How about `preparationId` and keep the rest as they are?

Reply:
Yes, something like
`perparedID` and `appliedID`
or, more lengthy, 
`preparationID` and `applicationID`
is good.

There must be a different functions for each (`_prepareID` and `_appliedID`).

Moreover, I would rename
```solidity
function _getPluginInstallationId(address _dao, address _plugin) pure returns (bytes32) {
    return keccak256(abi.encode(_dao, _plugin));
}
```
to `function _daoPluginId`